fix conflicts of audit rules for privileged commands

vojtapolasek opened the pull request:
#### Description:

- modify audit_rules_privileged_commands to unify with templated rules audit_rules_privileged_commands_*

    - remove checking for perm=x

    - remove remediationg of perm=x

    - unify checks and remediations to use -F key=privileged

#### Rationale:

The "automagic" rule audit_rules_privileged_commands and individual templated rules audit_rules_privileged_commands_*  were not cooperating together, although they are are often selected in one profile. It resulted in one of rules failing although the configuration was actually correct. It also resulted in duplicate Audit rule entries.

- fixes #5984

yuumasato commented:
@vojtapolasek So this is a regression after https://github.com/ComplianceAsCode/content/pull/5956?

vojtapolasek commented:
> @vojtapolasek So this is a regression after #5956?

Only partially. Yes the removal of perm=x was not introduced into the automagic rule, that's true. But additionally there was a problem that the templated rule and the automagic rule used different rule keys.

yuumasato reviewed (changes requested):
Test scenarios are failing because the script that generates the correct solution has not been updated.
https://github.com/ComplianceAsCode/content/blob/e5a65a1de7eda39e237401417196e87fbc6ea840/linux_os/guide/system/auditing/auditd_configure_rules/audit_privileged_commands/audit_rules_privileged_commands/tests/generate_privileged_commands_rule.sh#L7

And I suggest to add a new `fail` test scenario with audit rules with `-F perm=x`. To assure it removes the `perm` correctly.
In extension of previous PR, I suggest adding similar same test scenario for `audit_rules_privileged_commands_sudo`.

vojtapolasek commented:
You are right, I forgot to update tests. I fixed it and this led to discovering of a typo in oval. I also updated rule description. I will do separate update for templated audit_rules_privileged_commands rules.

yuumasato commented on shared/bash_remediation_functions/perform_audit_rules_privileged_commands_remediation.sh:
```diff
@@ -103,13 +103,12 @@ do
 		base_search=$(sed -e '/-a always,exit/!d' -e '/-F path='"${sbinary_esc}"'[^[:graph:]]/!d'		\
 				-e '/-F path=[^[:space:]]\+/!d'						\
 				-e '/-F auid>='"${min_auid}"'/!d' -e '/-F auid!=\(4294967295\|unset\)/!d'	\
+				-e '/-F perm=[rwxa]\+/d' \
```
@vojtapolasek Could you clarify what issue this is fixing?

yuumasato replied:
The `base_search` still needs to look for audit rules setting up a specific `-F perm=`.
But then, instead of removing the code block below (former lines 130 to 154) the block should be changed to remove the `perm` filter.

Test scenario `augenrules_rules_with_perm_x.fail.sh`  scenario is passing, but in the end it duplicates the audit rules:
```
-a always,exit -F path=/usr/bin/passwd -F perm=x -F auid>=1000 -F auid!=unset -k privileged
...
-a always,exit -F path=/usr/bin/passwd -F auid>=1000 -F auid!=unset -F key=privileged
```

yuumasato commented on shared/bash_remediation_functions/perform_audit_rules_privileged_commands_remediation.sh:
```diff
@@ -127,31 +126,6 @@ do
-				sed -i "s#${rule_head}\(.*\)${rule_tail}#${rule_head}${new_rule_perm}${rule_tail}#" "$afile"
-
-			fi
 
```
Instead of being removed, this block needs to be changed to remove any `-F perm=`.

vojtapolasek commented:
I modified the remediation. The base_search does not pick rules with perm=x explicitly, I don't think it makes sense here as they are covered by rest of regexes. I modified the remediation so that it removes the perm flag as requested.

yuumasato reviewed (approved):
Thank for addressing he comments, looks good to me.
I found a few typo issues, though.

yuumasato commented on shared/bash_remediation_functions/perform_audit_rules_privileged_commands_remediation.sh:
```diff
@@ -122,10 +122,10 @@ do
 			handled_sbinaries=("${handled_sbinaries[@]//-F path=/}")
 
-		# 		Merge the list of such SUID/SGID binaries found in this iteration with global list ignoring duplicates
+			# 		Merge the list of such SUID/SGID binaries found in this iteration with global list ignoring duplicates
```
```suggestion
			# Merge the list of such SUID/SGID binaries found in this iteration with global list ignoring duplicates
```
